flycheck.el

```diff
@@ -9480,6 +9480,26 @@ or if the current buffer has no file name."
   (and buffer-file-name
        (locate-dominating-file buffer-file-name "Cargo.toml")))
 
+(defun flycheck-rust-cargo-metadata ()
+  "Exec 'cargo metadata' and return the result as parsed JSON object.
+
+Return nil if the command failed or if a parse error occured."
+  (let ((metadata-lines
+         (ignore-errors
+           (process-lines "cargo" "metadata" "--no-deps" "--format-version" "1"))))
+    (and metadata-lines
+         (ignore-errors
+           (json-read-from-string (string-join metadata-lines ""))))))
+
+(defun flycheck-rust-cargo-workspace-root ()
+  "Return the path to the workspace root of a Rust Cargo project.
+
+Return nil if the workspace root does not exist (for Rust
+versions inferior to 1.25)."
+  (let ((metadata (flycheck-rust-cargo-metadata)))
+    (let-alist metadata
```

Comment:
No need to bind `metadata` here, I think (`(let-alist (flycheck-rust-cargo-metadata) …)`)